iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -31,13 +31,45 @@ UnixDomainSocket::UnixDomainSocket() noexcept
 }
 
 UnixDomainSocket::UnixDomainSocket(const std::string& name,
+                                   const IpcChannelMode mode,
+                                   const IpcChannelSide channelSide,
+                                   const size_t maxMsgSize,
+                                   const uint64_t maxMsgNumber) noexcept
+    : UnixDomainSocket(
+        NoPathPrefix,
+        [&]() -> std::string {
+            /// invalid names will be forwarded and handled by the other constructor
+            /// separately
+            if (!isNameValid(name))
+            {
+                return name;
+            }
+            return std::string(PATH_PREFIX) + name;
```

Comment:
Please don't return a `std::string` but a `cxx::string` instead

Reply:
This has to be done in a separate task when we get rid of `std::string` in our code base. It is not in the scope of this pull request to remove `std::string` from `UnixDomainSocket`.

Reply:
Got your point. Could you please create an new issue for this topic? Cheers!